inbm/trtl/factory/compose.go

```diff
@@ -73,8 +85,8 @@ func (compose *ComposeInfo) Login(username string, serverName string) {
 	// Dispatcher-agent, but because all commands are tied to the app type it will call the Compose Login instead
 	// of the Docker one.
 
-	if !is_username_registry_safe(username, serverName) {
-            fmt.Fprintf(os.Stderr, "Error: No special characters allowed in username/registry. List of good characters include: [a-z], [A-Z], [0-9], . , - , _, :")
+	if err := isRegistryCredentialsSafe(username, serverName); err != nil {
+            fmt.Fprintf(os.Stderr, err.Error())
```

Comment:
<h3>$\color{#FFA500}Improvements: Code Refactoring$</h3>

The `fmt.Fprintf(os.Stderr, err.Error())` pattern is used multiple times. Consider creating a utility function for error reporting to standardize error handling and make the code cleaner.